File: 1460-number-of-substrings-containing-all-three-characters/1460-number-of-substrings-containing-all-three-characters.cpp
class Solution {
public:
    int numberOfSubstrings(string s) {
        int l = 0;
        int r = 0;
        int hash[3] = {0};
        int count = 0;
        int n = s.size();
        while(r < n){
            hash[s[r] - 'a']++;
            while(hash[0] != 0 && hash[1]!=0 && hash[2] != 0){
                count += n-r;
                hash[s[l] - 'a']--;
                l++;
            }
            r++;
        }
        return count;
    }
};

File: 1460-number-of-substrings-containing-all-three-characters/1460-number-of-substrings-containing-all-three-characters.py
class Solution:
    def numberOfSubstrings(self, s: str) -> int:
        hash_array = [-1, -1, -1]
        count = 0
        for i in range(len(s)):
            hash_array[ord(s[i]) - ord('a')] = i
            if hash_array[0] != -1 and hash_array[1] != -1 and hash_array[2] != -1:
                count += min(hash_array) + 1
        
        return count
